Add new map "Skies of Land" by daveistaken

daveistaken opened the pull request:
![screenshot](https://github.com/MT-CTF/maps/assets/132380509/860ba7d7-a44b-4243-9092-c3bb3e4fbb51)

HobbitPower commented:
@daveistaken 
Haven't tested it yet, but I got a question: You only want to enable the map for classic? Not classes? (we can add that easily with editing the conf file later, if you want to enable your map for the classes mode)
You don't need stone sword and stone pick as an initial item, players get this anyway. So you can delete "ctf_melee:sword_stone" and "default:pick_stone" from line 12 of your conf file.
Also you can delete everything from """ (line 17) to """ (line 20) in your conf file. These are empty and unnecessary. 
You are new to Github so here's how to edit your conf file: 
1. go to daveistaken:master (so the master branch of your maps repo)
2. go to the skies_of_light_and_land folder and to your conf file there
3. click the edit symbol (the little pen)
4. edit it and click "commit changes"

daveistaken commented:
k, and aplogies if the screenshot wasn't centered properly in the 2:3 aspect ratio, i just couldn't find a perfect way to center it so i just cropped, and i couldn't think what license i'm supposed to use so i used the most common CC BY-SA 4.0

daveistaken commented:
alright the map.conf file is edited, and what you mentioned about the map only being supported for classic and not classes, it's because that some of the bases are vunlverable to being harmed by Magnum sniper rifles and some medics are able to just some gaps from a to b.

daveistaken commented:
*jump

HobbitPower commented:
> k, and aplogies if the screenshot wasn't centered properly in the 2:3 aspect ratio, i just couldn't find a perfect way to center it so i just cropped, and i couldn't think what license i'm supposed to use so i used the most common CC BY-SA 4.0

I can also take another screenshot and post it here, so you can change it later.
CC BY-SA 4.0 is fine, thats what we recommend. You can check this to learn more about it: https://creativecommons.org/about/cclicenses/

HobbitPower commented:
> alright the map.conf file is edited, and what you mentioned about the map only being supported for classic and not classes, it's because that some of the bases are vunlverable to being harmed by Magnum sniper rifles and some medics are able to just some gaps from a to b.

Alright, we will test the map later and we'll see ^^

daveistaken commented:
ok, i'll come at any time back soon.

src4026 commented:
The ores are perfect per team. I see that you have taken the decision of only enabling the map for classic and your reason is understandable. One thing that might make the game mechanics a bit difficult are the team chests that are placed far from the spawn. I could imagine a dire situation in which players need to get loot fast as approaching players are coming and wouldn't be able to get it in time. I would recommend you to place them closer to the flag/spawn for easier accessbility. Overall, its a map without any faults. I am looking forward to seeing more reviews on it :D.

daveistaken commented:
@src4026 I mostly accepted the question that you answered, However the last time I'd put the Chest Zones in the base itself it threw about 3 errors that there was a problem attempting to put 5 chests in just a single spot (1 managed to spawn), I'd moved it to two floating platforms and extended from 5 to 25 for more loot

src4026 commented:
@daveistaken I meant the team chests, not the normal ones.

daveistaken commented:
ah... Is there any chance I'd move the team chest from the inner base to outer base?

daveistaken commented:
@src4026 I'd altered the map.mts file

LuismiInTheZone commented:
We had the map ''abandoned islands'', and was deleted because the rate of suicide was up to the clouds, I don't want to happen the same with this

LuismiInTheZone commented:
By the way, this would be perfect for the new gamemode coming up called flag wars

LuismiInTheZone commented:
Also, I recommend you checking discord, it might be helpful if you need more info

daveistaken commented:
@LuismiInTheZone I'm too young to use Discord so I mostly use forums or GitHub instead

daveistaken commented:
still wip

daveistaken commented:
The map is modified so here the changelogs:
- Changed map name ("Skies of Light and Land" to "Skies of Land". Capsid suggested this)
- Chests now spawn underneath base instead of being in floating islands
- 2 islands with gold is removed and replaced with a bleak island (because if the other team focuses on the other team, the other team would snatch the ores and become more powerful)
- Changed skybox
- Fixed misaligned bases and barriers
- Increased map size (180x180?)

src4026 commented:
I'll take a look at the changes when I get back. Let's see what you have changed…

daveistaken commented:
> We had the map ''abandoned islands'', and was deleted because the rate of suicide was up to the clouds, I don't want to happen the same with this, but I like how is designed

Thanks, though since there's floating islands and if you fall into the ground you'll die because there's kill node on the ocean water (there used to be 4 stacks though reduced to 1)

a-blob commented:
Currently, there are a lot of problems with this map.

- You included team Red and Orange, which can cause some confusion since both teams have a similar color.
- I'm getting errors when loading the map, indicating that something is wrong with the chest zones. After loading the map, all teams have little/no treasure chests.
![image](https://github.com/MT-CTF/maps/assets/88883098/6c0a16e7-3286-4334-a444-2ab5f59d2537)

- The team zones are not correctly set. If I go up the stairs behind the flags and touch the barrier glass, I get teleported back to spawn.
- Since the distance between the teams isn't small, I find that the 60 blocks of cobble the players get when they spawn are not enough.

Interesting map sky map. I like the idea. However, because of the kill nodes at the bottom, I feel like the number of deaths is going to skyrocket. I also don't like the look of the map. The floating islands look "artificial" and not natural. The island looks like a square block of dirt floating in the air. The other islands look like an upside-down triangle. 
In its current state, I am against adding this map.

src4026 commented:
Looks like some of the changes dave made caused some errors.

daveistaken commented:
> Currently, there are a lot of problems with this map.
> 
> * You included team Red and Orange, which can cause some confusion since both teams have a similar color.
> * I'm getting errors when loading the map, indicating that something is wrong with the chest zones. After loading the map, all teams have little/no treasure chests.
>   ![image](https://user-images.githubusercontent.com/88883098/237562777-6c0a16e7-3286-4334-a444-2ab5f59d2537.png)
> * The team zones are not correctly set. If I go up the stairs behind the flags and touch the barrier glass, I get teleported back to spawn.
> * Since the distance between the teams isn't small, I find that the 60 blocks of cobble the players get when they spawn are not enough.
> 
> Interesting map sky map. I like the idea. However, because of the kill nodes at the bottom, I feel like the number of deaths is going to skyrocket. I also don't like the look of the map. The floating islands look "artificial" and not natural. The island looks like a square block of dirt floating in the air. The other islands look like an upside-down triangle. In its current state, I am against adding this map.

@a-blob You can add the lable "WIP" as I'll try to fix these bugs

daveistaken commented:
> Looks like some of the changes dave made caused some errors.

@src4026 unfortunately it did (:)

src4026 commented:
I took a look at the map. It shows the errors for flags and chests. You will need to re-add them properly. As for the map, why did you remove the chest islands? I think you should make the map bigger and include more islands. As GreenBlob mentioned, they seem artificial. You should probably consider making them more natural and resemble them close to "pieces of broken land" to add to the levitating effect. Adding more islands and details to the map will make it more interesting and unique. And also the Map Hint which you have changed to isn't giving any information about the hidden treasures. The previous one you had written was better than the present one. 

I think you should change these things:
1. Make the main islands larger *or* make the bottom tips petrude more. (or both)
2. If you will be making it larger, you should also probably make the map larger too. And along with that add minor islands as a helping hand for the journey to the flags but not too close.

daveistaken commented:
> I took a look at the once again map. It shows the errors for flags and chests. You will need to re-add them properly. As for the map, why did you remove the chest islands? I think you should make the map bigger and include more islands. As GreenBlob mentioned, they seem artificial at the moment. You should probably consider making them more natural and resemble them close to "pieces of broken land" to add to the levitating effect. Adding more islands and details to the map will make it moe interesting and unique. since after your removal of the chest islands, it seems quite empty. And also the Map Hint which you have changed to isn't giving any information about the hidden treasures. The previous one you had written was better than the present one.
> 
> I think you should change these things:
> 
> 1. Make the main islands larger _or_ make the bottom tips petrude more. (or both)
> 2. If you will be making it larger, you should also probably make the map larger too. And along with that add minor islands as a helping hand for the journey to the flags but not too close.
> 
> That's all I have for now 👍 .

@src4026 Right, the reason why I removed the chest islands every team cuz it's more way efficient for other teams to bridge from A to B, and every time you enter the exact map it slightly shifts the position of the map 1 block down and 3 blocks away from either X & Z, which is annoying that it takes me minutes to re-adjust them again

and there's some changes I'll do:
- Replace the void node to source lava in the floor of the map (to reduce the suicide rate, which is related to the old map 'abandoned islands')
- Move the gold source islands to the centre of the map (between the map and the barrier itself) and redesign the way it looks
- Redesign the lighthouse and the base
- Remove 2 teams from the map ✂
- Add more islands and make them bigger
- increase build distance from base to centre land from 45 to 60
- Change the map hint from 'Use blocks to bridge from A to B!' to 'Find a secret area that would lead you to some treasures...'
- Increase world size (it will take a long time (:)

this change might be too much (:)

src4026 commented:
> and every time you enter the exact map it slightly shifts the position of the map 1 block down and 3 blocks away from either X & Z, which is annoying that it takes me minutes to re-adjust them again

Hmm. What do you mean shifts the map? When?

daveistaken commented:
> > and every time you enter the exact map it slightly shifts the position of the map 1 block down and 3 blocks away from either X & Z, which is annoying that it takes me minutes to re-adjust them again
> 
> Hmm. What do you mean shifts the map? When?

@src4026 caused by the overlapping of the map being at the same location (the old one was mine) and it deicides to move in a different place to prevent overlapping ?

src4026 commented:
@daveistaken I am not able to understand what the exact process is that you do that leads to the shifting. As far as I know, that shouldn't be the case. Is that you open the world you were making the map in, and then reload your map from the CTF game maps list? Or something else?

daveistaken commented:
> @daveistaken I am not able to understand what the exact process is that you do that leads to the shifting. As far as I know, that shouldn't be the case. Is that you open the world you were making the map in, and then reload your map from the CTF game maps list? Or something else?

@src4026 got it now, so since the other file has loaded in the same location, i added the new file 'over' the old position like before and that might be why? (i wasn't able to understand the cause of the shifting)

src4026 commented:
I still have 0 clue of what you're talking about so I don't know :shrug:.

daveistaken commented:
> I still have 0 clue of what you're talking about so I don't know 🤷.

@src4026 wasn't able to understand anyway.

daveistaken commented:
did test it alone in minetest alone but couldn't test the master version (:)

MoreKat reviewed (commented):
I like the idea of having a sky map again. But I'm afraid this map in its current state will be too hard to capture because the islands are so far away from each other and on top of that you've slowed down the speed. Enemies can be seen from miles away and you have to build bridges to get anywhere. I think a solution would be to move the islands closer together (more to the center) and maybe add some prebuilt bridges.
The hint `In water islands, find breakable block for emergency ores, Instead. there's a secret area that leads you to dig more ores....` does not make much sense to me.  What are water islands? I understand you mean the brick blocks on the center island with 'breakable block'. I honestly don't think it's worth putting ores there unless it's diamonds or mese, so you might want to consider moving the mese that's currently on the islands with the white flags. If you've done that, I would rework the hint to `There are much more valuable ores to find beneath the lighthouse other than the ones in your mine.`
You should really focus on making the map not too hard to capture, since that's one of the reasons why the last sky map got removed. 

Also a question out of curiosity, what is this? (see picture)
<img width="183" alt="image" src="https://github.com/MT-CTF/maps/assets/73161488/1df4860e-4810-4cfc-8226-9011b556eac9">

MoreKat commented:
> they meant to be islands that are closer to each team with have 4 mese ores hidden in the breakable sand.

Oh I missed those, found only the ones on the islands with the white flags and the ores under the lighthouse. They're kind of very hard to find. Maybe specify which type of block to dig in your hint.

daveistaken commented:
@MoreKat concerning the fact that since the islands is too far away from the lighthouse, it's my 'only' choice to increase the distance from the lighthouse and slowing down the speed for journey's. however I'll have more time fixing some issues in sometime(Saturday, Friday?) that you spotted for it's current state of the map (my own)

daveistaken commented:
also I'm reluctant to tell everyone whatever which specific block is used to find the secret

daveistaken commented:
@MoreKat since that if i remove the mese from white falgs, how can i think of adding a diffent one?

MoreKat commented:
> @MoreKat since that if i remove the mese from white falgs, how can i think of adding a diffent one?

If I understand your question correctly, you could move the mese to the hiding spot in the center underneath the lighthouse instead.

daveistaken commented:
@MoreKat working in progress

daveistaken commented:
before i upload the files to mi file

i provide this screenshot
![image](https://github.com/MT-CTF/maps/assets/132380509/e29dbf42-29bb-4ddb-bc36-c496dddfa675)


and the changes i do:
Extended bridge from the central island for easy accessibility
Decreased blocks from spawn from 80 to 40 (half of them)
More islands are added for easily access
The sealant inside the islands is replaced by soild dirt instead of lava (Using the clipping glitch gets you inside inner if you manage to pass 1/2 of the frame)
The island is reworked except the base island is slightly unnatural
the base floor (the map itself) is replaced with the identical method from the 2nd lastest release (because you can escape from the lava with blocks and it takes you seconds to die, idc about the suicide rate anyway)
The lighthouse body is slightly more defensive (ex. The security fence is added because if the other person gets all the ores first they would conquer)

checks:
I checked the ores for the lighthouse:
24 iron
24 gold
12 diamond
tested the map and found issues and fixed them ?

except for the ores in the team base....

src4026 commented:
Everything looks good. Here are a few points that I would like to suggest:
- "There is ores inside the sand in the water island that is closer to your base, and there are much more valuable ores to find beneath the lighthouse other than the ones in your mine." to something like "There are ores inside the sand in the water island near to your base, and there are much more valuable ores to find beneath the lighthouse." The phrase comparing the ones in the mine and the ones in the mid, I feel, is not required and is simply making the hint long.

And some other notes:
- Tip: To make islands (the base, as you have mentioned previously) seem natural, TNT might be useful.
- I would recommend adding more islands _or_ increasing the distance between the central and the spawn islands and bringing the outer distant islands closer in. People won't really have motivation to cross the islands, resulting in only the central and chest islands being visited. I think adding bridges to the small islands would be better than adding a bridge making it easier to reach the central island, or adding bridges for the sub-islands along with the one for the central one.
> checks:
> I checked the ores for the lighthouse:
> 24 iron
> 24 gold
> 12 diamond
- By "gold", you mean mese, isn't it?
- I tried digging the sand islands with the ponds for the ores, but it seems that they are indestructible, and thus, inaccessible. Maybe I missed something, but I think the ores in them are supposed to be mined, isn't it?

I like how you have made the sub-islands look like :D. The white spiral you have added below the spawns are a really cool idea IMO. Perhaps adding lighting to it will make it more visible. Also, why not add it to the other sub-islands too? That's all I have for this review. Great work, but there is still room for improvement :+1:.  I really want this to be special, as it has potential to be added. I really miss the old sky island map and would love to see a new one.

daveistaken commented:
> Everything looks good. Here are a few points that I would like to suggest:
> 
> * "There is ores inside the sand in the water island that is closer to your base, and there are much more valuable ores to find beneath the lighthouse other than the ones in your mine." to something like "There are ores inside the sand in the water island near to your base, and there are much more valuable ores to find beneath the lighthouse." The phrase comparing the ones in the mine and the ones in the mid, I feel, is not required and is simply making the hint long.

i edit it later
> 
> And some other notes:
> 
> * Tip: To make islands (the base, as you have mentioned previously) seem natural, TNT might be useful.

wdym by 'TNT' ?
> * I would recommend adding more islands _or_ increasing the distance between the central and the spawn islands and bringing the outer distant islands closer in. People won't really have motivation to cross the islands, resulting in only the central and chest islands being visited. I think adding bridges to the small islands would be better than adding a bridge 

adding bridges to sub-islands is kind of a bad idea since if the player get the first first they sometimes put blocks to block the bridge to prevent the other player to stop the flag capturer even when trying to use the pistol included when you spawn, however i know that it was good idea to add it to reduce time of bridging from A to B

>making it easier to reach the central island, or adding bridges for the sub-islands along with the one for the central one.
> 
> > checks:
> > I checked the ores for the lighthouse:
> > 24 iron
> > 24 gold
> > 12 diamond
> 
> * By "gold", you mean mese, isn't it?
just a misspelled word

> * I tried digging the sand islands with the ponds for the ores, but it seems that they are indestructible, and thus, inaccessible. Maybe I missed something, but I think the ores in them are supposed to be mined, isn't it?

keep searching or there are breakable sand in each corner on the island
> 
> I like how you have made the sub-islands look like :D. The white spiral you have added below the spawns are a really cool idea IMO. Perhaps adding lighting to it will make it more visible. Also, why not add it to the other sub-islands too? That's all I have for this review. Great work, but there is still room for improvement 👍. I really want this to be special, as it has potential to be added. I really miss the old sky island map and would love to see a new one.

p.s: i have more time to edit in later days because i nearly worked on fixing for 4 days now (:)

daveistaken commented:
> @daveistaken
> 
> > wdym by 'TNT' ?
> 
> TNT as in the explosive TNT block in Minetest :). It's your wish how you want to structure the map in terms of the islands and sub-island bridges and distances, but remember that there should be active to promote the full usage of the map and not just only a part of it. Also, I will search once again for the breakable blocks 👍.

does tnt exist in the ctf game or it is useful to make some islands natural?

src4026 commented:
> > @daveistaken
> > 
> > > wdym by 'TNT' ?
> > 
> > TNT as in the explosive TNT block in Minetest :). It's your wish how you want to structure the map in terms of the islands and sub-island bridges and distances, but remember that there should be active to promote the full usage of the map and not just only a part of it. Also, I will search once again for the breakable blocks 👍.
> 
> does tnt exist in the ctf game or it is useful to make some islands natural? 
> 

TNT does in fact exist. I've used it to make caverns and to replicate natural deformities. I could perhaps show you an example of using it for the islands, but you will need to be careful not to blow the desired stuff.

daveistaken commented:
working in progress again (zzZZZ)

daveistaken commented:
> * Tip: To make islands (the base, as you have mentioned previously) seem natural, TNT might be useful.
> * I would recommend adding more islands _or_ increasing the distance between the central and the spawn islands and bringing the outer distant islands closer in. People won't really have motivation to cross the islands, resulting in only the central and chest islands being visited. I think adding bridges to the small islands would be better than adding a bridge making it easier to reach the central island, or adding bridges for the sub-islands along with the one for the central one.

I would know that adding bridges in the map would be fine because mobile couldn't place blocks well so i will add them

daveistaken commented:
@src4025 do you know a way to implement skyboxes into the map, i never tried to before

daveistaken commented:
i'll be inactive next day for 2 days cuz going to uk

src4026 commented:
> @src4025 do you know a way to implement skyboxes into the map, i never tried to before

I don't. I think this is the message you had sent me before contacting in-game :sweat_smile:. IIRC HobbitPower/Map Managers know about those or someone else who has worked with them.

src4026 commented:
> @src4025 do you know a way to implement skyboxes into the map, i never tried to before

~~src4026 :p~~

daveistaken commented:
@HobbitPower needed help with the backgrounds (:) ((for maps)

daveistaken commented:
added skybox for some purposes, made in 'Paint 3D'

daveistaken commented:
maybe, i try now at some time

p.s: I attempted to load the ctf master game but it threw an error indicating that the file for the Skybox mod is missing (init file)

daveistaken commented:
> @daveistaken take a look at this map: https://github.com/MT-CTF/maps/tree/master/capture_legend Edit: you already did this, my bad
> 
> Currently according to your conf file the skybox you are using is "TropicalSunnyDay". Since you will be using your custom skybox the line has to be changed to `skybox = skies_of_land` (so the name of this map folder).
> 
> For testing I suppose you need to put the skybox folder into your textures folder of maps (games>capturetheflag>[...]ctf_map>textures. And then you have to run this script if you are using linux: https://github.com/MT-CTF/capturetheflag/blob/master/scripts/setup_maps.sh Maybe @LoneWolfHT can confirm, if this is the correct way?

@HobbitPower the textures on the ctf_map folder is home to block textures instead of skybox textures for the map

LoneWolfHT commented:
> maybe, i try now at some time
> 
> p.s: I attempted to load the ctf master game but it threw an error indicating that the file for the Skybox mod is missing (init file)

You need to download CTF from ContentDB, unless you can use git

HobbitPower commented:
> @HobbitPower the textures on the ctf_map folder is home to block textures instead of skybox textures for the map

I'm aware of that, just remembered that gameit tested it that way and it worked for him. But yeah try it then with putting the skybox folder in the folder of your map and then running the script.

HobbitPower commented:
@daveistaken This is how it looks.
![image](https://github.com/MT-CTF/maps/assets/76528136/ebaceb8e-fba5-40d2-a6ec-42d7a27680a0)
Quite hard transitions at the edges of the image

daveistaken commented:
> @daveistaken This is how it looks. ![image](https://user-images.githubusercontent.com/76528136/247522085-ebaceb8e-fba5-40d2-a6ec-42d7a27680a0.png) Quite hard transitions at the edges of the image

this is just my quick way to make this drawing like this in 'Paint 3D' (obivusly a windows 10 progam)

HobbitPower commented:
> also how could i run the .Sh file script? (my operation sys is win10)

I also use windows, so I installed ubuntu for windows subsystem and read some instructions about how to run a .sh file with it online ;)

daveistaken commented:
@HobbitPower Didn't really need the .sh file because i noticed some of the skybox textures was missing is because it didn't exist in the Skybox mod (texture folder) included in the Capture the flag game, I can just copy some of the files and it would work ?

daveistaken commented:
> weird

@a-blob you meant the bottom of the skybox showing British Isles or it is the rest of the skybox itself ?

daveistaken commented:
> I still have 0 clue of what you're talking about so I don't know 🤷.

@s20 what i mean't by 'shifting' is that if I change the map size from it's original size, It throws a error for flags and chests previously because the map(width 2 & height) itself only uses the old size instead of a new size, causing to throw error if the flags and chest use new position instead of old position

src4026 commented:
> > I still have 0 clue of what you're talking about so I don't know shrug.
> 
> @s20 what i mean't by 'shifting' is that if I change the map size from it's original size, It throws a error for flags and chests previously because the map(width 2 & height) itself only uses the old size instead of a new size, causing to throw error if the flags and chest use new position instead of old position

Hmm. It's been long since that topic that I forgot everything :joy:

src4026 commented:
Skybox looks good :+1: . Nice!

daveistaken commented:
> Skybox looks good 👍 . Nice!

Grazie =)

a-blob commented:
> > weird
> 
> @a-blob you meant the bottom of the skybox showing British Isles or it is the rest of the skybox itself ?

Nevermind, seems like an interesting idea, we'll see later.

daveistaken commented:
not really sure if there's some unused block in the map 😑

daveistaken commented:
work in progress

daveistaken commented:
ready for review

HobbitPower commented:
It would be better if the screenshot shows the whole map (or at least most of it)

daveistaken commented:
> It would be better if the screenshot shows the whole map (or at least most of it)

io posta questa screenshot, perche il game no poteva screenshot a causa di problemi nel salvataggio della schermata.

i do it later

MoreKat commented:
@daveistaken please remove all references to real countries, including the British Isles in your skybox and the South African flag in the center of the map. This goes against our  general rule to avoid bringing up controversial topics such as politics. I'm sure you can come up with a cool-looking fictional flag (which cannot be mistaken for an existing flag). Also please take the screenshot without any texturepack enabled. This is not my final review.

daveistaken commented:
> @daveistaken please remove all references to real countries, including the British Isles in your skybox and the South African flag in the center of the map. This goes against our general rule to avoid bringing up controversial topics such as politics. I'm sure you can come up with a cool-looking fictional flag (which cannot be mistaken for an existing flag). Also please take the screenshot without any texturepack enabled. This is not my final review.

i'm removing them now

i thought adding these details should show a good look on the map, and this might be my first time violating the rules for the converstation topic (if they look down or at the flag)

right hwo i'll remove the down skybox and replace the african flag to lighthouse flag (The previous i added was the Irish flag)

daveistaken commented:
@Kat I might be worried that if i come up with a fictional flag (Which i'm not doing now) that if they are the same to a international flag I would get a strike for breaking the rules for the politics

daveistaken commented:
> @daveistaken please remove all references to real countries, including the British Isles in your skybox and the South African flag in the center of the map. This goes against our general rule to avoid bringing up controversial topics such as politics. I'm sure you can come up with a cool-looking fictional flag (which cannot be mistaken for an existing flag). Also please take the screenshot without any texturepack enabled. This is not my final review.

@MoreKat Those references that i added previously are now removed

MoreKat commented:
> > @daveistaken please remove all references to real countries, including the British Isles in your skybox and the South African flag in the center of the map. This goes against our general rule to avoid bringing up controversial topics such as politics. I'm sure you can come up with a cool-looking fictional flag (which cannot be mistaken for an existing flag). Also please take the screenshot without any texturepack enabled. This is not my final review.
> 
> @MoreKat Those references that i added previously are now removed

That's great, thanks. Please also use a screenshot with the default Minetest textures

daveistaken commented:
> > > @daveistaken please remove all references to real countries, including the British Isles in your skybox and the South African flag in the center of the map. This goes against our general rule to avoid bringing up controversial topics such as politics. I'm sure you can come up with a cool-looking fictional flag (which cannot be mistaken for an existing flag). Also please take the screenshot without any texturepack enabled. This is not my final review.
> > 
> > 
> > @MoreKat Those references that i added previously are now removed
> 
> That's great, thanks. Please also use a screenshot with the default Minetest textures

uhh the screenshot that i took is clearly default Minetest textures (look closely at islands), with blooming being added

HobbitPower commented:
@daveistaken Why did you build two red barriers instead of one in the center? Do you want to prevent people from getting to the centre too quickly? In my opinion there is no reason for that, because the bases are far enough apart anyway.

So please let me know your reasons, otherwise I would suggest to put a single red barrier in the middle. And then limit the barrier area (see conf file) to this area only. The more red barrier blocks there are, the greater delays there can be at the start of the game. That's why I'm surprised that your two red barriers consist of two layers each. One is perfectly sufficient.

daveistaken commented:
> @daveistaken Why did you build two red barriers instead of one in the center? Do you want to prevent people from getting to the centre too quickly? In my opinion there is no reason for that, because the bases are far enough apart anyway.
> 
> So please let me know your reasons, otherwise I would suggest to put a single red barrier in the middle. And then limit the barrier area (see conf file) to this area only. The more red barrier blocks there are, the greater delays there can be at the start of the game. That's why I'm surprised that your two red barriers consist of two layers each. One layer is perfectly sufficient.

@HobbitPower Two layer of each barrier Is used 'because they can throw or place items outside the border, and instead of entering the central island to just get loot, The hut is located at the left side of the base or either find ores in the well (water well) (there's 10 more chests than in the central island).

And if I remove both 2 layer and replace it with just 1 layer, Each team can get to access half of the chest to get loot, and the worst case If they're near the mine in the lighthouse (or either getting too good loot from chests), They'll just wait for the barrier to be finished to just get the ores from it' and kill enemy with stronger weapons they have if they try stand against a current enemy. 

Like you mentioned, One layer is sufficient, however I had to increase it to 3 layers of red barrier to just prevent them from seeing the mine or finding some valuable ores beyond. (the hidden secret is removed and easier to find)

Right now, I had to relax from editing the map to just fix a single' problem

daveistaken commented:
requested just a quick review

daveistaken commented:
> 
> many fences, basic base defence, slow general speed and 2 guarded islands are make to prevent the game from ending fast



> @daveistaken Why did you build two red barriers instead of one in the center? Do you want to prevent people from getting to the centre too quickly? In my opinion there is no reason for that, because the bases are far enough apart anyway.
> 
> So please let me know your reasons, otherwise I would suggest to put a single red barrier in the middle. And then limit the barrier area (see conf file) to this area only. The more red barrier blocks there are, the greater delays there can be at the start of the game. That's why I'm surprised that your two red barriers consist of two layers each. One layer is perfectly sufficient.

Well, i replaced the barrier with a single layer with one barrier in the center of the map anyways

src4026 commented:
Why have you made it such that no teams can access the ores before the match starts? Also, I cannot find the ores that you are referring to by the ones in the "sand wells". You have moved them to the cobblestone brick topped wells with stone and iron ores next to the mid-lighthouse, isn't it? If yes, then do please update the hint. The change of design you made looks cool IMO, and matches the light theme of the map (+ the uniformity now) :+1:. I am neutral on the addition of the prebuilt base structure.
Also, here's the count of ores, excluding iron (my bad for not checking it, I'll see to it later), in their respective zones (in the lighthouse):

Orange:
- 7 diamond ores
- 12 mese ores
Blue:
- 12 diamond ores
- 8 mese ores
Middle line:
- 3 diamond ore
- 2 mese ore

They are a bit unbalanced. 
Other than that, I am positive about the changes you've made in the previous couple of commits, though I will be having another look at the islands sometime later. Good :D

daveistaken commented:
> Why have you made it such that no teams can access the ores before the match starts?

Because if they have the ore pieces (from the lighthouse), It can be unbalanced quickly and the other team (chances) have no choice but to fight with iron ores provided in the mine or 13 mesa pieces from the sand well

> Also, I cannot find the ores that you are referring to by the ones in the "sand wells". You have moved them to the cobblestone brick topped wells with stone and iron ores next to the mid-lighthouse, isn't it? If yes, then do please update the hint. The change of design you made looks cool IMO, and matches the light theme of the map (+ the uniformity now) 👍. I am neutral on the addition of the prebuilt base structure. Also, here's the count of ores, excluding iron (my bad for not checking it, I'll see to it later), in their respective zones (in the lighthouse):
> 
> Orange:
> 
> * 7 diamond ores
> * 12 mese ores
>   Blue:
> * 12 diamond ores
> * 8 mese ores
>   Middle line:
> * 3 diamond ore
> * 2 mese ore
> 
> They are a bit unbalanced. Other than that, I am positive about the changes you've made in the previous couple of commits, though I will be having another look at the islands sometime later. Good :D

Hmmm, Didn't catch that the ores are unbalanced

src4026 commented:
@daveistaken also, the mese ores in the lighthouse seem to be indestructible. You'd want them to be destructible I suppose :P

daveistaken commented:
>Also, here's the count of ores, excluding iron (my bad for not checking it, I'll see to it later), in their respective zones (in the lighthouse):
> 
> Orange:
> 
> * 7 diamond ores
> * 12 mese ores
> 
> Blue:
> 
> * 12 diamond ores
> * 8 mese ores
> 
> Middle line:
> 
> * 3 diamond ore
> * 2 mese ore
> 
> They are a bit unbalanced. Other than that, I am positive about the changes you've made in the previous couple of commits, though I will be having another look at the islands sometime later. Good :D

for iron, there's 22 iron on blue side and 32 iron on orange side

MoreKat commented:
Where to find the mese ores? The ores underneath the lighthouse seem to be gone? I think having all these different spots with ores is really overcomplicating things. The hint is indeed unclear.

daveistaken commented:
> Where to find the mese ores? The ores underneath the lighthouse seem to be gone?

@MoreKat the ores still exists under the lighthouse, I just didn't make it clear

screenshot:
![screenshot_20230723_182011](https://github.com/MT-CTF/maps/assets/132380509/d1fc908a-bbeb-4f4a-a5d3-4de9d0a4874a)

src4026 commented:
How about something along the lines of:
>  …(Find the ores) in the (heart) of the lighthouse and the () wells next to it…

_*the above is just an outline that you can get ideas from_
It's your choice of your phrasing for the hint, but make sure it isn't too wordy and is to the point. I'll check your map later tomorrow :+1:.

src4026 reviewed (commented):
I feel this just made it longer. :thinking:. This can be made shorter IMO, I'll see to it later I guess.

src4026 commented on skies_of_land/map.conf:
```diff
@@ -0,0 +1,21 @@
+initial_stuff = return {"default:cobble 10","default:torch 5","default:stick 5","ctf_ranged:pistol_loaded 1"}
+author = daveistaken
+hint = Search the sandwell outside your base for ores buried in the sand. For even more valuable ores, check beneath in the heart of the lighthouse.
+license = CC BY-SA 4.0 | Hint is reworked with https://poe.com and it's edited by me.
```
```suggestion
license = CC BY-SA 4.0
```
I don't think it's needed. (*citation needed)

src4026 replied:
If you want to add it to the `Other Info`, then I guess it's your wish :thumbsup:

daveistaken replied:
I copied the hint to the website and then told AI to rework it (making it correct), also if that didn't exist in the license, it would be confusing if it was either written by human or 'artificial intelligence', surely it's necessary, but I won't remove it (just to give extra credit)

src4026 replied:
> @src4026 I copied the hint to the website and then told AI to rework it (making it correct), also if that didn't exist in the license, it would be confusing if it was either written by human or 'artificial intelligence', surely it's necessary, but I won't remove it (just to give extra credit)

> If generative AI output is considered a derivative work of the training materials, engineering teams that use it would be required to comply with the license(s) of the code upon which the tool is trained. This, of course, could come with requirements to disclose source code, generate attribution notices, and more.
> But this is only the case if generative AI output contains copyrightable expression. And, although the book isn’t closed on this question, there are considerations that suggest it doesn’t. Sufficiently short phrases — which generative AI often produces in the software development context — are not copyrightable and therefore neither subject to copyright nor any open source license.

From https://fossa.com/blog/generative-ai-and-software-development-copyright-law-and-license-compliance/

daveistaken commented:
@src4026 Couldn't find any information for citation

src4026 commented:
> @src4026 Couldn't find any information for citation

You can read this for reference:

> If generative AI output is considered a derivative work of the training materials, engineering teams that use it would be required to comply with the license(s) of the code upon which the tool is trained. This, of course, could come with requirements to disclose source code, generate attribution notices, and more.
> But this is only the case if generative AI output contains copyrightable expression. And, although the book isn’t closed on this question, there are considerations that suggest it doesn’t. Sufficiently short phrases — which generative AI often produces in the software development context — are not copyrightable and therefore neither subject to copyright nor any open source license.

From https://fossa.com/blog/generative-ai-and-software-development-copyright-law-and-license-compliance/

daveistaken commented:
> > @src4026 Couldn't find any information for citation
> 
> You can read this for reference:
> 
> > If generative AI output is considered a derivative work of the training materials, engineering teams that use it would be required to comply with the license(s) of the code upon which the tool is trained. This, of course, could come with requirements to disclose source code, generate attribution notices, and more.
> > But this is only the case if generative AI output contains copyrightable expression. And, although the book isn’t closed on this question, there are considerations that suggest it doesn’t. Sufficiently short phrases — which generative AI often produces in the software development context — are not copyrightable and therefore neither subject to copyright nor any open source license.
> 
> From https://fossa.com/blog/generative-ai-and-software-development-copyright-law-and-license-compliance/

@src4026 I'll sort it out tomorrow

HobbitPower commented on skies_of_land/map.conf:
```diff
@@ -0,0 +1,21 @@
+hint = Search the sandwell outside your base for ores buried in the sand. For even more valuable ores, check beneath in the heart of the lighthouse.
+license = CC BY-SA 4.0
+others = With assistance of @Kat, @GreenBlob, @HobbitPower and @s20 (for finding bugs and giving information), A floating map designed only for Classic.
+treasures = grenades:frag_sticky;1;1;2;0.1;1;
```
```suggestion
treasures =
```
The sticky grenades are not working properly afaik.  So remove that please.

daveistaken replied:
ok, but, after resolve this, Are the Frag sticky grenades in progress?

daveistaken commented:
work in progress ⛏

a-blob commented:
A lot seems to have improved since the last time I tested this. The map looks pretty good and the builds are nice. So far the ores seem balanced. It is now harder to fall off the islands thanks to the bridges, even though I still fall sometimes. Some of the island shapes still look a bit artificial to me but overall it is fine.
I like the idea of having a larger/taller structure in the middle of the map but now the lighthouse is removed. 
Here are some problems/suggestions I would like to point out: 

- During build-time, players get teleported back to base when they touch the glass barriers or the red build-time barriers. Staying on the edge of the team-zone teleports you back to base, so the team-zone should include the barriers. 
- I find the hint confusing. The punctuation is weird and I don't think the first letter following a comma needs to be in upper-case
- You did mention that "that some of the bases are vulnerable to being harmed by Magnum sniper rifles" so what is the idea behind adding the pistol to the default items? Personally, I don't think it is a very good idea since this map requires bridging, which makes the player who's bridging a good target for people shooting with guns. Savanna Ravine has this problem and sometimes it's a nightmare.

Overall I like the map 👍

daveistaken commented:
> * You did mention that "that some of the bases are vulnerable to being harmed by Magnum sniper rifles" so what is the idea behind adding the pistol to the default items? Personally, I don't think it is a very good idea since this map requires bridging, which makes the player who's bridging a good target for people shooting with guns. Savanna Ravine has this problem and sometimes it's a nightmare.
> 
> Overall I like the map 👍

The pistol was previously added for use as a emergency as a person tries steals the flag quickly to finish
Yet I'll remove the pistol as they can shoot everywhere they want (they can find the rifiles though, which cannot be disabled)

And how does the barrier work? (I normally put it outside the red barrier)

src4026 commented:
You have made several architectural changes over the course of the last few sets of commits. I am happy with them and the map now has a more consistent theme. The iron ores a balanced. However, you seem to have taken the decision to not keep mese/diamond ore on the map, while it being a map only for classic, if I did not miss anything.
As for the team zones, they seem to now work fine. I noticed that there's some part of the `Red Barrier Glass` that does not get removed. You could either increase the `Barrier Area` to include those particular blocks or remove them completely - you decide. The hint makes more sense now, but can be made shortened and clearer by just mentioning where they are/a hint to them (+ what GreenBlob has told above). Also, which area do you exactly mean by "mining area"? And, I seem to find only a little iron in the wells in the rest are in the other sand islands, but the islands are not destructible as per my observations. So please clarify this. Keep up the fantastic work :1:

src4026 commented:
> @src4026 Mining area might supposed to refer for the 'lighthouse' contain ores (but thats removed now), and don't know what else I can put some bigger mining areas in the tic-tac-toe styled islands (ツ ), also all of the islands are non-destructible as there is no emergency safely for the other platform, And mese do in fact exist in the map, via checking some certain chests (as a 5 percent chance)

Ah, so the mese can be found in chests. Also, do you have a rough count of the approximate "available" steel to the players?

daveistaken commented:
> > @src4026 Mining area might supposed to refer for the 'lighthouse' contain ores (but thats removed now), and don't know what else I can put some bigger mining areas in the tic-tac-toe styled islands (ツ ), also all of the islands are non-destructible as there is no emergency safely for the other platform, And mese do in fact exist in the map, via checking some certain chests (as a 5 percent chance)
> 
> Ah, so the mese can be found in chests. Also, do you have a rough count of the approximate "available" steel to the players? Seems to be only a handful and not enough for a small team

@src4026 There's 52 iron (can be burned into steel) for red and blue, Though I don't have the rough count of the approximate "available" steel to the player (like 104 iron-steel in total)

daveistaken commented:
they can also find steel swords in chests

src4026 commented:
> > > @src4026 Mining area might supposed to refer for the 'lighthouse' contain ores (but thats removed now), and don't know what else I can put some bigger mining areas in the tic-tac-toe styled islands (ツ ), also all of the islands are non-destructible as there is no emergency safely for the other platform, And mese do in fact exist in the map, via checking some certain chests (as a 5 percent chance)
> > 
> > Ah, so the mese can be found in chests. Also, do you have a rough count of the approximate "available" steel to the players? Seems to be only a handful and not enough for a small team
> 
> @src4026 There's 52 iron (can be burned into steel) for red and blue, Though I don't have the rough count of the approximate "available" steel to the player (like 104 iron-steel in total)

Are you sure they _all_ are available?

daveistaken commented:
> > > > @src4026 Mining area might supposed to refer for the 'lighthouse' contain ores (but thats removed now), and don't know what else I can put some bigger mining areas in the tic-tac-toe styled islands (ツ ), also all of the islands are non-destructible as there is no emergency safely for the other platform, And mese do in fact exist in the map, via checking some certain chests (as a 5 percent chance)
> > > 
> > > 
> > > Ah, so the mese can be found in chests. Also, do you have a rough count of the approximate "available" steel to the players? Seems to be only a handful and not enough for a small team
> > 
> > 
> > @src4026 There's 52 iron (can be burned into steel) for red and blue, Though I don't have the rough count of the approximate "available" steel to the player (like 104 iron-steel in total)
> 
> Are you sure they _all_ are available?

y e s

src4026 commented:
Oh, I just realised my mistake.

daveistaken commented:
> Oh, I just realised my mistake. Sorry for that.

thats okay (theek hai)

MoreKat reviewed (commented):
I've got a few remarks:
1. Please change the hint so it includes the location of the majority of the ores (the mine down the ladder). I don't think you even have to mention the 'sandwell' (quite unclear, a made up word) because there are barely any ores there. Also IMO it's better to say 'mese crystals spawn in chests'.
2. I suggest adding more blocks to initial_stuff, since people will have to build bridges. 10 is not sufficient.
3. You should add coal to the aforementioned mine. Currently there is only one coal to find, unless I am wrong, which I very well could be. You cannot smelt 51 iron ingots with one coal. Having to use wood for this is a bit inconvenient.
4. I think you should decrease the rarity of the mese (so increase the probability), just a bit. Right now very few mese crystals spawn. 
5. This is largely a personal opinion but the random capital letter in the middle of a sentence bothers me in others: `With assistance of @Kat, @GreenBlob, @HobbitPower and @s20 (for finding bugs and giving information), A floating map designed only for Classic.` Correct would be: `... information), a floating ...` 

Other than that, I find the map very cool and I look forward to having a map with sky islands again.

daveistaken commented:
@MoreKat for 3 (of what you said) I couldn't just put the mixup of iron and coal ore (With Worldedit) so i ignored it, they can still use sticks but is it still sufficient?

daveistaken commented:
also i might include bridges for each base (because of mobile players)

MoreKat commented:
> @MoreKat for 3 (of what you said) I couldn't just put the mixup of iron and coal ore (With Worldedit) so i ignored it, they can still use sticks but is it still sufficient?

You can also replace the 4 iron ore in the wall behind the base with coal. Or put them in another place if you can't put them in that mine.

daveistaken commented:
> > @MoreKat for 3 (of what you said) I couldn't just put the mixup of iron and coal ore (With Worldedit) so i ignored it, they can still use sticks but is it still sufficient?
> 
> You can also replace the 4 iron ore in the wall behind the base with coal. Or put them in another place if you can't put them in that mine.

@MoreKat I already put the coal in the mine (where the ladder leads to), so meaning there approximately 59 coal ores, 59/65 iron ore, in one part

HobbitPower reviewed (approved):
Good looking sky map, approved 👍

daveistaken commented:
@MoreKat, @HobbitPower Thx for the merge (i tried to be patient), btw, if that voide node is increasing the suicide rate, just tell me to change it
